app/handlers/base.py

```diff
@@ -305,6 +325,25 @@ def error(self, message, data={}, status=400, extra={}):
         self.set_status(status)
         self.write({"status": "error", "message": message, "data": data, **extra})
 
+        if self.current_user is None:
+            return
+        with DBSession() as session:
+            session.merge(self.current_user)
+            if hasattr(self.current_user, "created_by_id"):
+                user_id = int(self.current_user.created_by_id)
+            else:
+                user_id = int(self.current_user.id)
+
+            api_call = APICall(
+                user_id=user_id,
+                method=self.request.method,
+                uri=self.request.uri.split("?")[0],
+                size=sizeof(data),
+                success=False,
```

Comment:
I think if you stick something like `self.start_time = time.time()` up in the `prepare` method, you could also get the runtime of the query. That should work unless there's async stuff happening, in which case who knows what will happen... but it is worth trying out and comparing to timing data you get from the side calling the API.